numpy/array_api/tests/test_creation_functions.py

```diff
@@ -124,3 +126,11 @@ def test_zeros_like_errors():
     assert_raises(ValueError, lambda: zeros_like(asarray(1), device="gpu"))
     assert_raises(ValueError, lambda: zeros_like(asarray(1), dtype=int))
     assert_raises(ValueError, lambda: zeros_like(asarray(1), dtype="i"))
+
+def test_meshgrid_dtype_errors():
+    # Doesn't raise
+    meshgrid()
```

Comment:
A bit surprising:
```
>>> np.meshgrid()
[]
```
But ah well, that is what it is.

Reply:
I don't see why that's unexpected. `meshgrid(*args)` should return a list of the same length as `args`. I don't know if it's ever useful. Possibly if there is code that passes an unknown set of arrays fo meshgrid it can be the correct null condition.

More importantly though I think if functions with `*args` can produce meaningful and consistent results with empty `args`, even if the result is itself something trivial like an empty list, that is preferable to raising an exception. That way the function actually matches its signature. Consider something like `result_type(*arrays_and_dtypes)` which is a bit misleading because `result_type()` does not actually work.